### PR TITLE
Windows: Fix compiling cpp with tree artifact as source

### PR DESCRIPTION
**Pull request by meteorcloudy**
Fixes https://github.com/bazelbuild/bazel/issues/9321

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java [hunk 1]
```diff
@@ -1398,12 +1398,16 @@ private Artifact createCompileActionTemplate(
             /* additionalBuildVariables= */ ImmutableMap.of()));
     semantics.finalizeCompileActionBuilder(configuration, featureConfiguration, builder);
     // Make sure this builder doesn't reference ruleContext outside of analysis phase.
+    SpecialArtifact dotdTreeArtifact = null;
```
Comment (ulfjack) [inline src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java]: Can you add a comment here what this is for?

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileActionTemplate.java [hunk 2]
```diff
@@ -127,9 +127,12 @@
         TreeFileArtifact outputTreeFileArtifact =
             ActionInputHelper.treeFileArtifactWithNoGeneratingActionSet(
                 outputTreeArtifact, PathFragment.create(outputName), artifactOwner);
-        TreeFileArtifact dotdFileArtifact =
-            ActionInputHelper.treeFileArtifactWithNoGeneratingActionSet(
-                dotdTreeArtifact, PathFragment.create(outputName + ".d"), artifactOwner);
+        TreeFileArtifact dotdFileArtifact = null;
+        if (dotdTreeArtifact != null) {
```
Comment (ulfjack) [inline src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileActionTemplate.java]: Can you please also mark dotdTreeArtifact as @Nullable?

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java (hunk 1 above)
Comment (meteorcloudy) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CcCompilationHelper.java]: Done

File: src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileActionTemplate.java (hunk 2 above)
Comment (meteorcloudy) [reply src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileActionTemplate.java]: Thanks, done!

**Pull request closed without merging**